newsroom/agenda/agenda.py

```diff
@@ -386,91 +386,131 @@ def _set_event_date_range(search):
 
     should = []
 
+    # Scenario 1: Default query with only date_from
     if date_from and not date_to:
-        # Filter from a particular date onwards
-        should = gen_date_range_filter("dates.end", "gte", search.args.get("date_from"), date_from)
+        should.extend(
+            [
+                # Events that start after or on date_from
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_from}}},
+                            {"range": {"dates.end": {"gte": date_from}}},
+                        ],
+                        "must_not": {"term": {"dates.all_day": True}},
+                    }
+                },
+                # All-day events starting or ending after date_from
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_from}}},
+                            {"range": {"dates.end": {"gte": date_from}}},
+                        ],
+                        "filter": {"term": {"dates.all_day": True}},
+                    }
+                },
+            ]
+        )
+
+    # Scenario 2 and 3: Query with specific filters
     elif not date_from and date_to:
-        # Filter up to a particular date
-        should = gen_date_range_filter("dates.end", "lte", search.args.get("date_to"), date_to)
+        # Apply date range filters for next_7_days, next_30_days, etc.
+        should.extend(
+            [
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_range["gt"], "lte": date_range["lt"]}}},
+                        ],
+                        "must_not": {"term": {"dates.all_day": True}},
+                    }
+                },
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_range["gt"], "lte": date_range["lt"]}}},
+                        ],
+                        "filter": {"term": {"dates.all_day": True}},
+                    }
+                },
+            ]
+        )
+
     elif date_from and date_to:
-        # Filter based on the date range provided
-        should = [
-            {
-                # Both start/end dates are inside the range
-                "bool": {
-                    "filter": [
-                        {"range": {"dates.start": {"gte": date_from}}},
-                        {"range": {"dates.end": {"lte": date_to}}},
-                    ],
-                    "must_not": {"term": {"dates.all_day": True}},
+        # Filter based on both date_from and date_to
+        should.extend(
+            [
+                # Events that start and end within the range
```

Comment:
what we actually want here is events that stared before the end and their end date is after the start,
so:
```py
{"must": [
    {"range": {"dates.start": {"lte": date_to}}},
    {"range": {"dates.end": {"gte": date_from}}},
]}
```
should cover all the cases, we just need to have also the all_day version